OneCommunityGlobal/HighestGoodNetworkApp
# Tianyuan_fix_email_list_visibility

# Description
<img width="739" alt="截屏2023-11-08 18 55 28" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/73833864/5f2e9213-b099-437e-97be-f33feb9213ec">


## Related PRS (if any):
No

## Main changes explained:
- Pass the state "auth" from WeeklySummaryReport component to FormattedReport component in order to get the account role 
- Create a new component EmailsList to contain all the behaviors related to the email list  
- Program the logical code to make only Admin and Owner have access to the email list 
…

## How to test:
1. check into current branch
2. do `npm install` and `npm run start:local` to run this PR locally
3. log as admin/owner/mentor/manger/core team user (they are the only five roles have acess to the WeeklySummaryReport page)
4. go to dashboard→ Reports→ WeeklySummaryReport→ waiting for the page loading -> after loading, scoll to the bottom of the page 
5. Test this PR in two steps:
If you log as **admin/owner**, make sure you can have the access to see the email list
If you log as **mentor/manger/core team**, make sure you can **NOT** see the email list 

## Screenshots or videos of changes:
In this video, the console displays the role of account, but in this PR I have commented the console log code to print the role of account. **So you don't need to open your console.** 


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/73833864/9fd7d4c5-9538-4663-b1b0-bd3da66a6144


## Note:
Include the information the reviewers need to know.
